### PR TITLE
migration: Add case for poweroff vm

### PR DESCRIPTION
**Pull request by cliping**
VIRT-294690 - VM live migration - poweroff vm during migration

Signed-off-by: cliping <lcheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            test.fail("Migrated VM failed to be in %s state at source" % vm_state_src)
+        test.log.info("Guest state is '%s' at source is as expected", vm_state_dest)
+        migration_obj.verify_default()
+
+    test_case = params.get('test_case', '')
+    vm_name = params.get("migrate_main_vm")
+
+    vm = env.get_vm(vm_name)
+    migration_obj = base_steps.MigrationBase(test, vm, params)
+    setup_test = eval("setup_%s" % test_case) if "setup_%s" % test_case in \
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py:65]: I am a little confused about logic here, what purpose setup_test wants to achieve here. 
The same apply for verify_test as well

**Review by dzhengfy — APPROVED** [review]
LGTM

File: libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/guest_lifecycle_operations_during_migration/migration_poweroff_vm.py:65]: @chunfuwen This is to get the real function name to be used for this case, like `setup_poweroff_vm`, then this function could be invoked implicitly in runtime.

**Comment by cliping** [comment]
Test result:
 (1/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.with_precopy.p2p: PASS (175.76 s)
 (2/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.with_precopy.non_p2p: PASS (169.34 s)
 (3/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.with_postcopy.p2p: PASS (326.82 s)
 (4/4) type_specific.io-github-autotest-libvirt.migration.guest_lifecycle_operations_during_migration.migration_poweroff_vm.with_postcopy.non_p2p: PASS (359.37 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**